### PR TITLE
Better error message when input is not valid jsonl

### PR DESCRIPTION
**Pull request by om-sharma**
The app throws an error when the input entry is not a valid jsonl. The error response was 500 status and the error message was `{"code":500,"message":"HTTP 500 Internal Server Error"}`. It shouldn't be 500 response because the error was because of wrong input entry and also the error message was not very helpful.

Now we are returning much better error with 400 error response.

### CODE REVIEW

**Comment by philandstuff** [comment]
It's not great to refer to a private trello card from a public github PR. Can you include the example inline?

File: app/src/main/java/uk/gov/mint/ExceptionUtils.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.mint;
+
+class ExceptionUtils {
+    @SuppressWarnings("unchecked")
+    private static <T extends Exception> void throwAsUnchecked(Throwable exception) throws T {
+        throw (T) exception;
```
Comment (philandstuff) [inline app/src/main/java/uk/gov/mint/ExceptionUtils.java:6]: I've seen this trick before, it's clever but also evil :smiling_imp:

File: app/src/main/java/uk/gov/mint/LoadHandler.java [hunk 2]
```diff
@@ -39,7 +38,7 @@ private void processEntries(String[] entries) {
 //                        entryValidator.validateEntry(register, jsonNode);
                         return canonicalJsonMapper.writeToBytes(hashedEntry(jsonNode));
                     } catch (Exception ex) {
-                        throw Throwables.propagate(ex);
+                        return ExceptionUtils.rethrow(ex);
```
Comment (philandstuff) [inline app/src/main/java/uk/gov/mint/LoadHandler.java:43]: can you document why you're doing this sneaky throw here?

File: app/src/test/java/uk/gov/functional/FunctionalTest.java [hunk 3]
```diff
@@ -73,7 +73,7 @@ private void send(List<String> payload) {
         Response response = jerseyClient.target("http://localhost:4568/load")
                 .request(MediaType.APPLICATION_JSON_TYPE)
                 .post(Entity.json(String.join("\n", payload)));
-        if (response.getStatus() != 200) {
+        if (response.getStatus() != 204) {
```
Comment (philandstuff) [inline app/src/test/java/uk/gov/functional/FunctionalTest.java:76]: why did this change? (and should the test maybe just ensure it's in the 2xx range?)

**Comment by om-sharma** [comment]
Updated the description!!

File: app/src/test/java/uk/gov/functional/FunctionalTest.java (hunk 3 above)
Comment (om-sharma) [inline app/src/test/java/uk/gov/functional/FunctionalTest.java:76]: The response must be 204. Earlier  we were returning 200 which was wrong.

**Comment by philandstuff** [comment]
what is the error message in the new response?

**Comment by om-sharma** [comment]
The error message would be something like `Error parsing json input: <result of e.getMessage() from com.fasterxml.jackson.core.JsonParseException>`.

**Comment by philandstuff** [comment]
here's the exact message I get:

```
$ curl -isd '{
> }' localhost:4567/load ; echo
HTTP/1.1 400 Bad Request
Date: Fri, 18 Dec 2015 16:03:06 GMT
Content-Type: text/plain
Content-Length: 178

Error parsing json input: Unexpected end-of-input: expected close marker for OBJECT (from [Source: [B@45434920; line: 1, column: 0])
 at [Source: [B@45434920; line: 1, column: 3]
```

EDIT: it's ugly but at least it's a 400 not a 500 now.

**Pull request merged**